### ISSUE TITLE
AIX machines are all offline

### ISSUE THREAD

**Issue by MylesBorins** [nodejs/build#2387]
https://ci.nodejs.org/job/node-test-commit-aix/nodes=aix71-ppc64/

**Comment by richardlau** [comment]
According to the logs both agents crashed:

test-ibm-aix71-ppc64_be-1:
```
$ tail -25 jenkins_console.log
INFO: Connecting to ci.nodejs.org:41913
Jul 11, 2020 12:32:48 AM hudson.remoting.jnlp.Main$CuiListener status
INFO: Trying protocol: JNLP4-connect
Jul 11, 2020 12:32:54 AM hudson.remoting.jnlp.Main$CuiListener status
INFO: Remote identity confirmed: 4f:22:06:c2:64:29:f3:3f:6d:7c:d7:01:f2:18:bb:93
Jul 11, 2020 12:32:54 AM hudson.remoting.jnlp.Main$CuiListener status
INFO: Connected
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGDANGER (0x21) at pc=0x090000000054a8c4, pid=12845126, tid=0x0000000000000203
#
# JRE version: OpenJDK Runtime Environment (8.0_192-b12) (build 1.8.0_192-b12)
# Java VM: OpenJDK 64-Bit Server VM (25.192-b12 mixed mode aix-ppc64 compressed oops)
# Problematic frame:
# C  [libpthread.a+0xffffffff]  _event_sleep+0x584
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /home/iojs/hs_err_pid12845126.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
#
$
```

test-ibm-aix71-ppc64_be-2:
```
$ tail -25 jenkins_console.log
INFO: Connecting to ci.nodejs.org:41913
Jul 20, 2020 2:05:34 AM hudson.remoting.jnlp.Main$CuiListener status
INFO: Trying protocol: JNLP4-connect
Jul 20, 2020 2:05:39 AM hudson.remoting.jnlp.Main$CuiListener status
INFO: Remote identity confirmed: 4f:22:06:c2:64:29:f3:3f:6d:7c:d7:01:f2:18:bb:93
Jul 20, 2020 2:05:39 AM hudson.remoting.jnlp.Main$CuiListener status
INFO: Connected
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGDANGER (0x21) at pc=0x090000000054a8c4, pid=5832896, tid=0x0000000000000203
#
# JRE version: OpenJDK Runtime Environment (8.0_192-b12) (build 1.8.0_192-b12)
# Java VM: OpenJDK 64-Bit Server VM (25.192-b12 mixed mode aix-ppc64 compressed oops)
# Problematic frame:
# C  [libpthread.a+0xffffffff]  _event_sleep+0x584
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /home/iojs/hs_err_pid5832896.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
#
$
```

I've restarted the agent on both test machines, via:

```
sudo /etc/rc.d/rc2.d/S20jenkins start
```

**Comment by richardlau** [comment]
They're both offline again 😞.

**Comment by richardlau** [comment]
The agents crashed again. `SIGDANGER` looks like an AIX mechanism for warning when the paging space is low: https://www.ibm.com/support/knowledgecenter/en/ssw_aix_72/devicemanagement/page_space_trouble.html

The most obvious thing to have recently changed is the recent Jenkins server upgrade (https://github.com/nodejs/build/issues/2384). I've updated the agents (via the [`ansible/playbooks/jenkins/worker/upgrade-jar.yml`](https://github.com/nodejs/build/blob/08131cf63cd94bc9979c75eaf60923d1bc5ccb97/ansible/playbooks/jenkins/worker/upgrade-jar.yml) playbook) in case this is somehow related and restarted them.

Lets keep this open until the AIX test machines have been online for more than a day. If we're still seeing crashes my next thing to try would be to update the version of Java 8 that we're using -- we're currently using https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/tag/jdk8u192-b12 which is from October 2018 (https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/tag/jdk8u262-b10 is the current Java 8).
https://github.com/nodejs/build/blob/9cf3680add6ee833cff9bdf855d5c9ab70c1532a/ansible/roles/java-base/tasks/main.yml#L124-L129

**Comment by mhdawson** [comment]
Given the Java versions are from 2018, I'd agree updating the Java version would make sense.

**Comment by AshCripps** [comment]
the agent also crashed on the new arm macs after the jenkins upgrade with a similar crash message

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGBUS (0xa) at pc=0x000000010ea81d19, pid=60537, tid=0x0000000000003e03
#
# JRE version: OpenJDK Runtime Environment (8.0_262-b10) (build 1.8.0_262-b10)
# Java VM: OpenJDK 64-Bit Server VM (25.262-b10 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# V  [libjvm.dylib+0x280d19]
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /Users/iojs/hs_err_pid60537.log
#
# Compiler replay data is saved as:
# /Users/iojs/replay_pid60537.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
#
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  Internal Error (safepoint.cpp:384), pid=60537, tid=0x0000000000002f03
#  guarantee(PageArmed == 0) failed: invariant
#
# JRE version: OpenJDK Runtime Environment (8.0_262-b10) (build 1.8.0_262-b10)
# Java VM: OpenJDK 64-Bit Server VM (25.262-b10 mixed mode bsd-amd64 compressed oops)
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /var/folders/39/wd0dfrtn6dz7ljd73n32rxtw0000gp/T//hs_err_pid60537.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
#
# [ timer expired, abort... ]
```

**Comment by richardlau** [comment]
Both AIX test machines are still online, so it looks like updating the agent has fixed this. I'll raise a separate issue for updating Java on these machines as it looks like we'd want to update on other platforms as well.

@AshCripps if updating the agent on the arm macs doesn't fix those I suggest opening a new issue or tracking under the existing issues for Apple silicon.
